testing/scripts/test_operator_updates.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+
+from seldon_e2e_utils import (
+    initial_rest_request,
+    rest_request,
+    retry_run,
+    wait_for_status,
+    wait_for_rollout,
+)
+
+
+def assert_model(sdep_name, namespace, initial=False):
+    _request = initial_rest_request if initial else rest_request
+    r = _request(sdep_name, namespace)
+
+    assert r is not None
+    assert r.status_code == 200
+    assert r.json()["data"]["tensor"]["values"] == [1.0, 2.0, 3.0, 4.0]
+
+    retry_run(f"kubectl get -n {namespace} sdep {sdep_name}")
+
+
+@pytest.mark.sequential
+@pytest.mark.parametrize(
+    "seldon_version",
+    [
+        pytest.param("0.4.1", marks=pytest.mark.skip(reason="fixed in Helm 3.0.3")),
+        "0.5.1",
+        "1.0.0",
+        "1.0.1",
+    ],
+    indirect=True,
+)
+def test_cluster_update(namespace, seldon_version):
+    # Deploy test model
+    retry_run(f"kubectl apply -f ../resources/graph1.json -n {namespace}")
+    wait_for_status("mymodel", namespace)
+    wait_for_rollout("mymodel", namespace)
+    assert_model("mymodel", namespace, initial=True)
+
+    # Upgrade to source code version cluster-wide.
+    # Note that this upgrade should leave the cluster as it was before the
+    # test.
+    # TODO: There is currently a bug updating from 0.4.1 using Helm 3.0.2.
+    # This will be fixed once https://github.com/helm/helm/pull/7269 is
+    # released in Helm 3.0.3.
+    retry_run(
```

Comment:
I think this would be relevant for another test as these test is not for rolling update, but for just generally updates, but maybe it would be good to add another issue for rolling updates where we can send continuous requests to show that the model can be updated without interrupting requests (as right now it uses initial=True for both tests)

Reply:
I see cluster-wide updates as a subset of rolling updates. We need to ensure a safe upgrade path between versions (as in, nothing obvious breaking like missing webhooks or resources) in order to enable more advanced update models.

On the second point, I agree that a good extra for all these tests would be making sure that there is no downtime whenever we upgrade SC (using any of the update options). I'll check which changes would be required and if it's not too much I think we can add it as part of this PR (otherwise I'll create a separate issue).

Reply:
Good point on first point. Cool stuff, this PR looks solid, probably good to go so we can add that to a followup